Source/WebCore/html/HTMLButtonElement.cpp

```diff
@@ -249,4 +249,14 @@ bool HTMLButtonElement::isExplicitlySetSubmitButton() const
     return isSubmitButton() && hasAttributeWithoutSynchronization(HTMLNames::typeAttr);
 }
 
+bool HTMLButtonElement::isKeyboardFocusable(KeyboardEvent* event) const
+{
+    return Element::isKeyboardFocusable(event);
+}
+
+bool HTMLButtonElement::isMouseFocusable() const
+{
+    return isFocusable() && !!tabIndexSetExplicitly();
+}
+
```

Comment:
I wonder if we should just change the logic at https://searchfox.org/wubkat/rev/c36758395b16531cacdf1dbdb1315a45c992ca87/Source/WebCore/html/HTMLFormControlElement.cpp#243-259 instead?

`<input type=button|submit|reset>` should behave the same as `<button>`

Reply:
I now singled out button and <input type=button|submit|reset> to allow keyboard/mouse focus if tabindex is set, I don't think we want more form controls with the new behavior than that?